lib/internal/flutter_ble_lib_mixins/discovery_mixin.dart

```diff
@@ -0,0 +1,60 @@
+part of flutter_ble_lib;
+
+mixin DiscoveryMixin on FlutterBLE {
+  Future<void> discoverAllServicesAndCharacteristics(
+      Peripheral peripheral, String transactionId) async {
+    await _methodChannel.invokeMethod(
+        MethodName.discoverAllServicesAndCharacteristics, <String, dynamic>{
+      ArgumentName.deviceIdentifier: peripheral.identifier,
+      ArgumentName.transactionId: transactionId
+    });
+  }
+
+  Future<List<Service>> services(Peripheral peripheral) async {
+    String jsonString = await _methodChannel.invokeMethod(
+        MethodName.services, <String, dynamic>{
+      ArgumentName.deviceIdentifier: peripheral.identifier
+    });
+
+    List<String> decodedJson = (jsonDecode(jsonString) as List<dynamic>).cast();
```

Comment:
Is it necessary to cast it co List of strings? I think we could just leave it as List of dynamics. jsonDecode accepts whatever you passes to it.

Reply:
I need to cast those dynamics to String before parsing it. `jsonDecode(String sourceString...`